OpenRA.Game/Traits/World/ScreenMap.cs

```diff
@@ -117,9 +120,14 @@ public void Update(IEffect effect, WPos position, Sprite sprite)
 			Update(effect, position, size);
 		}
 
-		public void Remove(IEffect e)
+		public void Remove(IEffect effect)
 		{
-			partitionedEffects.Remove(e);
+			partitionedEffects.Remove(effect);
+		}
+
+		bool ValidBounds(Rectangle bounds)
+		{
+			return bounds.Width != 0 && bounds.Height != 0;
```

Comment:
Can be simplified to `bounds != Rectangle.Empty` or maybe changed to `bounds.Width > 0 && bounds.Height > 0`.